### PR TITLE
- check private imports

### PR DESCRIPTION
**Pull request by SSE4**
related https://github.com/conan-io/hooks/issues/176

check for private APIs, disallowed examples:
```
from conans.model.version import Version
from conans.client.tools import msvs_toolset
```
allowed examples:
```
from conans.tools import Version
from conans.errors import ConanInvalidConfiguration
```

### CODE REVIEW

**Review by danimtb — APPROVED** [review]

**Comment by SSE4** [comment]
docs: https://github.com/conan-io/conan-center-index/pull/2662

**Comment by uilianries** [comment]
That's a great hook! At least two times I needed to restrict it in CCI during a review!

File: hooks/conan-center.py [hunk 1]
```diff
@@ -552,6 +553,32 @@ def test(out):
                           'version "{}".'.format(version, conandata_path, config_path, config_path,
                                                  version))
 
+    @run_test("KB-H053", output)
+    def test(out):
+        def _is_private_import(line):
+            allowed_list = ["tools", "errors"]
+            for pattern in ["from conans.", "import conans."]:
```
Comment (uilianries) [inline hooks/conan-center.py:560]: well, I can import conans and call conans.user.something :smiling_imp:

Comment (SSE4) [reply hooks/conan-center.py:560]: I don't think we should go too deep, just prevent simple cases.
I still can do alias imports and other hacks, but that's beyond the goal.

Comment (uilianries) [reply hooks/conan-center.py:560]: I agree!

**Review by jgsogo — APPROVED** [review]
Like it. It will hit some recipes and they will become better recipes 😄

**Comment by jgsogo** [comment]
@SSE4 we need to fix a conflict here

**Comment by jgsogo** [comment]
Now a different conflict, after reverting KB053, nevertheless I'll add this one as KB054 (so we don't modify https://github.com/conan-io/conan-center-index/blob/master/docs/error_knowledge_base.md#kb-h053-invalid-symlinks), and we will see if we can merge back KB053 again

**Comment by SSE4** [comment]
what? I've just solved conflict 15 minutes ago... how come it keeps conflicting every time?

File: hooks/conan-center.py [hunk 2]
```diff
@@ -50,7 +50,9 @@
              "KB-H049": "CMAKE WINDOWS EXPORT ALL SYMBOLS",
              "KB-H050": "DEFAULT SHARED OPTION VALUE",
              "KB-H051": "DEFAULT OPTIONS AS DICTIONARY",
-             "KB-H052": "CONFIG.YML HAS NEW VERSION"
+             "KB-H052": "CONFIG.YML HAS NEW VERSION",
+             "KB-H053": "PRIVATE IMPORTS",
+             "KB-H054": "INVALID SYMLINKS"
```
Comment (jgsogo) [inline hooks/conan-center.py:55]:
```suggestion
             "KB-H053": "INVALID SYMLINKS"
             "KB-H054": "PRIVATE IMPORTS",
```

Right now it should go in this ordering, unless we wait after the _symlinks_ issue is resolved

File: hooks/conan-center.py [hunk 3]
```diff
@@ -50,7 +50,9 @@
              "KB-H049": "CMAKE WINDOWS EXPORT ALL SYMBOLS",
              "KB-H050": "DEFAULT SHARED OPTION VALUE",
              "KB-H051": "DEFAULT OPTIONS AS DICTIONARY",
-             "KB-H052": "CONFIG.YML HAS NEW VERSION"
+             "KB-H052": "CONFIG.YML HAS NEW VERSION",
+             "KB-H053": "INVALID SYMLINKS"
```
Comment (jgsogo) [inline hooks/conan-center.py:54]:
```suggestion
             "KB-H053": "INVALID SYMLINKS",
```

**Review by jgsogo — APPROVED** [review]

**Pull request merged**